### PR TITLE
Type script usability

### PR DESCRIPTION
**Pull request by BoBiene**

### CODE REVIEW

File: src/H5PEditor.ts
Comment (sr258) [inline src/H5PEditor.ts:89]: Could you please add this new parameter to docs/h5p-editor-constructor.md? I think it would be great if you explained what a typical use case for using a custom URL generator is.

File: src/types.ts
Comment (sr258) [inline src/types.ts]: This import seems to be unused.

**Review by sr258 — CHANGES_REQUESTED** [review]
Thank you for your PR! 

I have two small change requests (adding the new parameter of H5PEditor to the documentation and removing an unused import) and would like to understand your use case better: I see that you want to improve subclassing the file content storage classes and this is fine with me. Can you explain your use case in which the protected getter is necessary instead of simply accessing the field directly? (Same with the base url in the UrlGenerator.)

**Comment by sr258** [comment]
@BoBiene I'd be willing to merge this, if you address the two issues I've mentioned and explained why you need the protected getters.

**Comment by BoBiene** [comment]
@sr258 Sure, thank you. 
I'm working on them. 
Currently I have a working prototype of my implementation and as soon as it is all on place I will update this Pullrequest.

File: src/adapters/expressErrorHandler.ts
Comment (sr258) [inline src/adapters/expressErrorHandler.ts]: Was this removed deliberately? I think we still need it to switch the language to the current user's settings. I have already wondered whether it might not be better to have the implementation call changeLanguage and remove the language override. Any thoughts on this?

If we remove the languageOverride, the currying function in l. 16 must be removed as well, as it doesn't make any sense otherwise.

File: src/adapters/expressController.ts
Comment (sr258) [inline src/adapters/expressController.ts]: I think you need to pass in the mimetype of the file, but you pass in the filename. I receive warnings in the browser's dev console because of this and some stylesheets are not rendered correctly. 

We already use the npm package mime-types, so you can use it to determine the mimetypes.

File: src/adapters/expressController.ts
Comment (sr258) [inline src/adapters/expressController.ts]: See above.

**Review by sr258 — CHANGES_REQUESTED** [review]
Thank you for you extensive PR! I think it indeed improved the developer friendliness of the project a lot, especially for newcomers! 

I've taken the liberty to directly correct a few documentation issues and push the changes. The problem with the mimetypes must be resolved before it can be merged and you should double-check whether you really wanted to remove the language switch in the Express error handler.

File: src/adapters/expressErrorHandler.ts
Comment (BoBiene) [reply src/adapters/expressErrorHandler.ts]: It was a mistake while fixing the merge-conflict.
I will correct this.

File: src/adapters/expressController.ts
Comment (BoBiene) [reply src/adapters/expressController.ts]: i see. fixed that.

**Review by sr258 — APPROVED** [review]
I think everything is in order now. Thank you again!

**Comment by BoBiene** [comment]
@JPSchellenberg & @sr258 thank you.

**Pull request merged**
